CryptoTechReminderSystem.Test/SlackGatewayTests.cs

```diff
@@ -15,91 +17,145 @@ public class SlackGatewayTests
+            public void CanSendAMessageToAUserWithAText()
+            {
+                var text = "Please make sure your timesheet is submitted by 13:30 on Friday.";
+                var message = new Message()
+                {
+                    Channel = "U0112WTW",
+                    Text = text
+                };
+            
+                _slackGateway.Send(message);
+            
+                var receivedRequest = _fluentSimulator.ReceivedRequests.First();
+                
+                JObject.Parse(receivedRequest.RequestBody)["text"].ToString().Should().Be(text);
+            }
+        }
+
+        [TestFixture]
+        public class CanRetrieveDevelopers
         {
-            var text = "Please make sure your timesheet is submitted by 13:30 on Friday.";
-            var message = new Message()
+            private FluentSimulator _fluentSimulator;
+            private SlackGateway _slackGateway;
+            private IList<SlackDeveloper> _response;
+
+            [SetUp]
+            public void SetUp()
             {
-                Channel = "U0112WTW",
-                Text = text
-            };
+                _fluentSimulator = new FluentSimulator(Address);
+                _slackGateway = new SlackGateway(Address, Token);
             
-            _slackGateway.Send(message);
+                var json = File.ReadAllText(
+                    Path.Combine(
+                        AppDomain.CurrentDomain.BaseDirectory,
+                        "../../../SlackUsersExampleResponse.json"
+                    )
+                );
+                
+                _fluentSimulator.Get("/api/users.list").Responds(json);
+            
+                _fluentSimulator.Start();
+
+                _response = _slackGateway.RetrieveDevelopers();
+            }
             
-            var receivedRequest = _fluentSimulator.ReceivedRequests.First();
+            [TearDown]
+            public void TearDown()
+            {
+                _fluentSimulator.Stop();
+            }
+            
+            [Test]
+            public void CanSendAGetUsersRequest()
+            {
+                var receivedRequest = _fluentSimulator.ReceivedRequests.First();
+                
+                receivedRequest.Url.Should().Be(Address + "api/users.list");
+            }
+            
+            [Test]
+            public void CanSendAGetUsersRequestWithAToken()
+            {
+                var receivedRequest = _fluentSimulator.ReceivedRequests.First();
                 
-            DeserializeObject<PostMessageRequest>(receivedRequest.RequestBody).Text.Should().Be(text);
+                receivedRequest.Headers["Authorization"].Should().Be("Bearer " + Token);
+            }
+        
+            [Test]
+            public void CanGetAListOfSlackDevelopers()
+            {
+                _response.Should().BeOfType<List<SlackDeveloper>>();
+                _response.Should().HaveCount(6);
```

Comment:
Seeing this without any given, when, then type syntax feels a little weird

Reply:
It is weird but this is only testing the Slack gateway and is not an acceptance test, so I think it is fine.

Reply:
Fair point